cli/src/pcluster/models/imagebuilder.py

```diff
@@ -112,47 +128,53 @@ def __init__(self, image_name: str, config: dict = None, stack: ImageBuilderStac
         self.image_name = image_name
         self.__source_config = config
         self.__stack = stack
+        self.__image = None
         self.__config = None
         self.__bucket = None
         self.template_body = None
         self._s3_artifacts_dict = {
             "root_directory": "parallelcluster",
-            "root_image_directory": "imagebuilders",
-            "source_config_name": "image-config-original.yaml",
+            "root_image_directory": "images",
             "config_name": "image-config.yaml",
             "template_name": "aws-parallelcluster-imagebuilder.cfn.yaml",
             "custom_artifacts_name": "artifacts.zip",
         }
         self.__s3_artifact_dir = None
 
     @property
-    def s3_artifacts_dir(self):
+    def s3_artifact_dir(self):
         """Get s3 artifacts dir."""
         if self.__s3_artifact_dir is None:
-            self._get_artifact_dir()
+            self.__s3_artifact_dir = self._get_artifact_dir()
         return self.__s3_artifact_dir
 
+    @property
+    def config_url(self):
+        """Return configuration file S3 bucket url."""
+        return self.bucket.get_config_url(self._s3_artifacts_dict.get("config_name"))
+
     @property
     def stack(self):
         """Return the ImageBuilderStack object."""
         if not self.__stack:
             try:
-                self.__stack = ImageBuilderStack(AWSApi().instance().cfn.describe_stack(self.image_name))
+                self.__stack = ImageBuilderStack(AWSApi.instance().cfn.describe_stack(self.image_name))
             except AWSClientError:
                 raise ImageBuilderActionError(f"ImageBuilder stack {self.image_name} does not exist.")
         return self.__stack
 
     @property
-    def source_config(self):
-        """Return config used to create the imagebuilder stack."""
-        if not self.__source_config:
+    def image(self):
+        """Return Image object."""
+        if not self.__image:
             try:
-                self.__source_config = self.bucket.get_config(config_name=self._s3_artifacts_dict.get("config_name"))
-            except Exception as e:
-                raise ImageBuilderActionError(
-                    "Unable to load configuration from bucket " f"'{self.bucket.name}/{self.s3_artifacts_dir}'.\n{e}"
-                )
-        return self.__source_config
+                self.__image = AWSApi.instance().ec2.describe_image_by_name_tag(self.image_name)
+            except AWSClientError as e:
+                if isinstance(e, ImageNotFoundError):
+                    raise ImageBuilderActionError(f"Image {self.image_name} does not exist.")
```

Comment:
I think you can use a different exception here: e.g. NonExistingImageError, given that you are catching it internally.

You can catch this new exception type in a different way, rather than looking at the error message.

Reply:
Added ImageError and subclass NonExistingImageError to imagebuilder.py. For all the places calling image property, we will catch ImageError and then check if it is instance of NonExistingImageError or not. Therefore, calling image property won't raise ImageBuilderActionError, instead raise ImageBuilderActionError in the caller.